.github/workflows/rock-update.yaml

```diff
@@ -0,0 +1,110 @@
+name: Update ROCK on new releases of its source
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+    inputs:
+      rock-name:
+        description: "Name of the application for which to build the ROCK"
+        required: true
+        type: string
+      source-repo:
+        description: "Repository of the source application in 'org/repo' form"
+        required: true
+        type: string
+      check-go: 
+        description: "Flag to check updates on the Go version"
+        default: false
+        required: false
+        type: boolean
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  check-version:
+    name: Detect new releases
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo snap install jq
+          sudo snap install yq
+
+      - id: latest-release
+        name: Fetch version used in *latest* release
+        run: |
+          TAG=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ inputs.source-repo }}/releases/latest \
+            | jq -r .tag_name)
+          echo "release=$TAG" >> $GITHUB_OUTPUT
+
+      - name: Checkout the ROCK source
+        uses: actions/checkout@v3
+        with:
+          path: main
+
+      - id: check
+        name: Check if the release has already been added
+        shell: bash
+        run: |
+          source_tag="${{ steps.check.outputs.release }}"
+          version=${source_tag#"v"}
+          if [ ! -f /$version/rockcraft.yaml ]; then
+            echo "version=$version" >> $GITHUB_OUTPUT
+            echo "release=${{steps.latest-release.outputs.release}}" >> $GITHUB_OUTPUT
+            echo "New upstream release ${{steps.latest-release.outputs.release}} found"
+          else
+            echo "No new upstream release found"
+          fi
+
+      - id: check-go-version
```

Comment:
Seems like this task is a no-op.

Reply:
It's just wrong naming, this is doing the checkout of the upstream application repository; fixing it

Reply:
We can probably drop the id field.